zip-0227.rst

```diff
@@ -1,8 +1,384 @@
+
+The protocol was designed with three main properties in mind:
+
+- At the time of issuance, the issuer can already allocate all the tokens to the corresponding owners by creating the corresponding (shielded) output notes to the respective addresses. As is implied, the issuance mechanism is itself transparent, but the allocation is totally private.
+- In a single issuance bundle, the issuer can publish many issuance actions, or in other words, create multiple custom Assets.
+- Every issuance action contains the ``finalize`` boolean that defines whether that specific custom Asset can have further tokens issued or not.
+
+The issuance mechanism is broad enough for issuers to either create Assets on Zcash (i.e. Assets that originate on the Zcash block chain), as well as for institutions to create bridges from other chains and import Wrapped Assets. In fact, the protocol described below enables what we hope will be a useful set of applications. The design decision is not to have a chosen name to describe the custom Asset, but to delegate it to an off-chain mapping, as this would imply a land-grab “war”.
+
+Issuance Keys
+-------------
+
+There are two keys relevant to the Issuer:
+
+1. The issuance authorizing key, denoted as :math:`\mathsf{isk}`, is the key that is used to sign the issuance transaction. This key is used to authorize the issuance of a specific Asset type, and is only used by the issuer. Throughout we also call this key the issuer signature key.
+
+2. The issuance validating key, denoted as :math:`\mathsf{ik}`, is the key that is used to validate the issuance transaction. This key is used to validate the issuance of a specific Asset type, and is used by all block chain users (specifically the Asset owners and consensus validators) to associate the Asset in question with the issuer. Throughout we also call this key the issuer verification key.
+
+The two keys are derived in an analogous manner to the spend authorizing key and spend validating key, using the same signature scheme, as described in ZIP 32 [#zip-0032]_ and in accordance with ZIP 316 [#zip-0316]_.
+
+- The issuance authorizing key is derived directly from the spending key, :math:`\mathsf{sk}`, as a private signature key:
+
+.. math:: \mathsf{isk := ToScalar^{Orchard}(︀ PRF^{expand}_{sk} ([0x0a])}
+
+- The issuance validating key is derived from the issuance authorizing key, :math:`\mathsf{isk}`, as a public verification key:
+
+.. math:: \mathsf{ik := SpendAuthSig^{Orchard}.DerivePublic(isk)}
+
+This allows the issuer to use the same wallet it usually uses to transfer Assets, while keeping a disconnect from the other keys. Specifically, this method is aligned with the requirements and motivation of ZIP 32 [#zip-0032]_. It provides further anonymity and the ability to delegate issuance of an Asset (or in the future, generate a multi-signature protocol) while the rest of the keys remain in the wallet safe.
```

Comment:
We may encourage versioning in the issuance key derivation, otherwise we are bound to this parameter set (the Pallas curve) forever

// @str4d

Reply:
Issuance key derivation has changed to be similar to ZIP 32, based on discussions in our syncs.